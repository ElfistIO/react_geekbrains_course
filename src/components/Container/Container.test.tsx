// import React from "react";
// import { fireEvent, render, screen } from "@testing-library/react";
// import "@testing-library/jest-dom";
// import { Container } from "./Container";
// import userEvent from "@testing-library/user-event";

// describe("Container", () => {
//   it("render component", () => {
//     render(<Container />);
//   });
//   it("render with snapshot", () => {
//     const { asFragment } = render(<Container />);
//     expect(asFragment()).toMatchSnapshot();
//   });
//   it("message test", () => {
//     render(<Container />);
//     fireEvent.input(screen.getByTestId("message"), {
//       target: { value: "Some message" },
//     });
//     fireEvent.click(screen.getByTestId("button"));
//     expect(screen.getByTestId("message")).toContainHTML("Some message");
//   });
//   it("bot answer", async () => {
//     render(<Container />);
//     fireEvent.input(screen.getByTestId("Bot"), {
//       target: { value: "Bot" },
//     });
//     await userEvent.click(screen.getByRole("button"));
//     expect(
//       setTimeout(() => screen.findByText("Your feedback accepted."), 1000)
//     ).toBeTruthy();
//   });
// });
