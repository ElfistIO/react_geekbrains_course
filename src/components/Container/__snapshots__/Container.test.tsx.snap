// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Container render with snapshot 1`] = `
<DocumentFragment>
  <div
    class="container"
  >
    <ul
      class="chat"
    />
    <form
      class="form"
    >
      <div
        class="MuiFormControl-root MuiTextField-root name__input css-1u3bzj6-MuiFormControl-root-MuiTextField-root"
      >
        <label
          class="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined MuiFormLabel-root MuiFormLabel-colorPrimary Mui-focused css-1sumxir-MuiFormLabel-root-MuiInputLabel-root"
          data-shrink="true"
          for="outlined-basic"
          id="outlined-basic-label"
        >
          Name
        </label>
        <div
          class="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary Mui-focused MuiInputBase-formControl css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
        >
          <input
            aria-invalid="false"
            class="MuiOutlinedInput-input MuiInputBase-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
            id="outlined-basic"
            placeholder="Enter your Name..."
            type="text"
            value=""
          />
          <fieldset
            aria-hidden="true"
            class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
          >
            <legend
              class="css-1in441m"
            >
              <span>
                Name
              </span>
            </legend>
          </fieldset>
        </div>
      </div>
      <div
        class="MuiFormControl-root MuiTextField-root feedback css-1u3bzj6-MuiFormControl-root-MuiTextField-root"
      >
        <div
          class="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl MuiInputBase-multiline css-dpjnhs-MuiInputBase-root-MuiOutlinedInput-root"
        >
          <textarea
            aria-invalid="false"
            class="MuiOutlinedInput-input MuiInputBase-input MuiInputBase-inputMultiline css-1sqnrkk-MuiInputBase-input-MuiOutlinedInput-input"
            id="outlined-multiline-flexible"
            placeholder="Type your feedback here..."
            rows="6"
            style="height: 0px; overflow: hidden;"
          />
          <textarea
            aria-hidden="true"
            class="MuiOutlinedInput-input MuiInputBase-input MuiInputBase-inputMultiline css-1sqnrkk-MuiInputBase-input-MuiOutlinedInput-input"
            readonly=""
            style="visibility: hidden; position: absolute; overflow: hidden; height: 0px; top: 0px; left: 0px; transform: translateZ(0); padding: 0px; width: 100%;"
            tabindex="-1"
          />
          <fieldset
            aria-hidden="true"
            class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
          >
            <legend
              class="css-hdw1oc"
            >
              <span
                class="notranslate"
              >
                â€‹
              </span>
            </legend>
          </fieldset>
        </div>
      </div>
      <button
        class="MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButtonBase-root Mui-disabled  css-sghohy-MuiButtonBase-root-MuiButton-root"
        disabled=""
        tabindex="-1"
        type="submit"
      >
        submit
      </button>
    </form>
  </div>
</DocumentFragment>
`;
